### PR TITLE
XS✔ ◾ Do you know how to indicate mandatory vs suggested PR changes?

### PR DESCRIPTION
**Pull request by danielmackay**
>
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ https://github.com/SSWConsulting/SSW.SAW.General/issues/36

> 2. What was changed?

✏️ Added rule

> 3. Did you do pair or mob programming (list names)?

✏️ No

### CODE REVIEW

File: rules/mandatory-vs-suggested-pr-changes/rule.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+---
+seoDescription: TODO
```
Comment (matt-goldman-ssw) [inline rules/mandatory-vs-suggested-pr-changes/rule.md:2]: Does this need attention before we publish?

File: rules/mandatory-vs-suggested-pr-changes/rule.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+type: rule
+title: Do you know how to indicate mandatory vs suggested PR changes?
+uri: mandatory-vs-suggested-pr-changes
+authors:
+  - title: Daniel Mackay
+    url: https://www.ssw.com.au/people/daniel-mackay
+  - title: Matt Goldman
+    url: https://www.ssw.com.au/people/william-liebenberg
```
Comment (matt-goldman-ssw) [inline rules/mandatory-vs-suggested-pr-changes/rule.md:10]:
```suggestion
    url: https://www.ssw.com.au/people/goldie
```

File: rules/mandatory-vs-suggested-pr-changes/rule.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+created: 2024-10-28T17:00:00.000Z
+guid: 6F5AE6FC-F29A-4938-B012-FDE7F669998C
+---
+
+When reviewing another developers pull request, review comments can fall into a mix of categories. Some comments might be required, while others might be optional. This is where the mandatory vs suggested PR changes rule comes in.
```
Comment (matt-goldman-ssw) [inline rules/mandatory-vs-suggested-pr-changes/rule.md:15]:
```suggestion
When reviewing another developer's pull request, review comments can fall into a mix of categories. Some changes might be required, while others might be optional. Do you know the best way to tell the requester which is which?
```

File: rules/mandatory-vs-suggested-pr-changes/rule.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+<!--endintro-->
+
+## Types of Pull Request Feedback
+
+When reviewing pull request comments can often be divided into two categories:
```
Comment (matt-goldman-ssw) [inline rules/mandatory-vs-suggested-pr-changes/rule.md:21]:
```suggestion
When reviewing pull requests, comments can often be divided into two categories:
```

File: rules/mandatory-vs-suggested-pr-changes/rule.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+* **Change** - you must change this if you want me to approve.
```
Comment (matt-goldman-ssw) [inline rules/mandatory-vs-suggested-pr-changes/rule.md:23]:
```suggestion
* **Change** - you must change this if you want me to approve
```

**Review by matt-goldman-ssw — CHANGES_REQUESTED** [review]
One questions and some suggestions.

Also I think we need something about the PBI comment. For this, there needs to be a link to the PBI. A "good Samaritan" approach is to create the PBI and link to it (add a `TODO`). Alternatively if you just leave the comment and want the requester to do it, they should do the same (add a `TODO` with a link to the PBI) before you approve it.

File: rules/mandatory-vs-suggested-pr-changes/rule.md (hunk 1 above)
Comment (danielmackay) [reply rules/mandatory-vs-suggested-pr-changes/rule.md]: Sorted

**Review by matt-goldman-ssw — APPROVED** [review]
🚀 LGTM

**Pull request merged**